### ISSUE TITLE
Using Sommelier as proxy

### ISSUE THREAD

**Issue by eadwu** [udevbe/greenfield#142]
Is it possible to have Sommelier as a proxy for a windows to Greenfield? Greenfield looks like it can maybe be used to make wslg far more usable?

The only real condition is for Greenfield to expose its Wayland display via a socket (though they seem to be addressed as URLs here) I believe?

**Comment by eadwu** [comment]
Should be possible, need to figure out some snags with WSL2 though.

**Comment by eadwu** [comment]
Having trouble running any `Remote Applications`.

I can't get any wayland application (`gtk4-demo`, `kwrite`), trying both the `docker-compose` and `yarn build`.

On Xwayland (`xterm`), I get an issue regarding failing to fetch the buffer (grep'ing doesn't fetch anything similar to this error), however it works fine on the `docker-compose`.

On WSL I've been playing around on the default Ubuntu distribution.

**Comment by Zubnix** [comment]
Hi, could you describe what you are trying to do as it's not entirely clear from your description. Are you simply trying to run it in wsl or something more?

**Comment by eadwu** [comment]
Yep, right now I'm simply trying to run it in WSL2. I managed to get everything working besides the Compositor Proxy.

**Comment by eadwu** [comment]
Managed to get the proxy working with docker after booting up a Linux VM and every intermediate output, turns out a file got corrupted for Docker on WSL and reinstalling the distro and packages fixed it.

Also managed to reproduce the buffer error with Docker, just switched `debian:bookworm-20231030-slim` to `debian:unstable` and the Docker proxy no longer worked on my end, so that explains that probably some package bump in Ubuntu probably caused it not to work outside of Docker.

Going to try to figure out what needs to be pinned / compiled from source so it can be run outside of Docker.

**Comment by eadwu** [comment]
Got it working e2e on Linux, nothing needed to be pinned, just `LIBGL_ALWAYS_SOFTWARE=1` ... `d3d12` causes problems apparently.

I can get it working from a Linux browser but if I try to connect from Windows, I can load the webpage and run the command, but no window is created on the browser on the Window's side.

**Comment by Zubnix** [comment]
I'll see if I can get something running and documented in the coming weeks. [Running on WSL used to work](https://github.com/udevbe/greenfield/issues/109) a couple of versions ago, so it should still work but perhaps some things have changed in the meantime.
